From: Hendrik Brueckner <brueckner@linux.vnet.ibm.com>
Subject: kernel: kernel module cannot be loaded
Patch-mainline: v3.9-rc1
Git-commit: e80cfc31d872b6b85b8966bce6ba80bee401a7dd
References: bnc#813708, LTC#91552

Description:  kernel: kernel module cannot be loaded
Symptom:      Particular kernel modules cannot be loaded.  Loading the
              modules fails with "Invalid module format".
Problem:      When building a kernel module, the module might contain
              R_390_NONE relocations.  The module loader does not know
              about these type of relocations and, hence, refuse to
              load these kernel modules.
Solution:     Add the R_390_NONE relocation type.
Reproduction: Create a kernel module that contains R_390_NONE
              relocations and load the kernel module.

Signed-off-by: Hendrik Brueckner <brueckner@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 arch/s390/kernel/module.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/s390/kernel/module.c
+++ b/arch/s390/kernel/module.c
@@ -203,6 +203,8 @@ apply_rela(Elf_Rela *rela, Elf_Addr base
 	val = symtab[r_sym].st_value;
 
 	switch (r_type) {
+	case R_390_NONE:	/* No relocation.  */
+		break;
 	case R_390_8:		/* Direct 8 bit.   */
 	case R_390_12:		/* Direct 12 bit.  */
 	case R_390_16:		/* Direct 16 bit.  */
